### PR TITLE
improvement (join-dedi): change label name button join-dedi and create method open questionnaire page

### PR DESCRIPTION
**Pull request by doohanas**
## Changes

- improvement (join-dedi): change label name button join-dedi and create method open questionnaire page 

## Preview

- Change label name button join-dedi and create method open questionnaire page 

   https://user-images.githubusercontent.com/79241754/178637043-bac21df5-bc16-4d43-84f8-05ee99b56585.mp4

## Evidence
title: improvement (join-dedi): change label name button join-dedi and create method open questionnaire page
project: Desa Digital
participants: @doohanas @yoslie @agunghide

### CODE REVIEW

File: components/Competition/Banner/index.vue [hunk 1]
```diff
@@ -36,6 +36,12 @@ export default {
   methods: {
     onClickCTA () {
       this.$emit('onClickCTA', true)
+    },
+    showQuestionnaire () {
+      /**
+       * Trigger to open questionnaire page.
+       */
+      this.$router.push({ path: '/registration', query: { option: 'desa' } })
```
Comment (yoslie) [inline components/Competition/Banner/index.vue:44]: in my opinion, we can just leave the function to emit onClickCTA on each components section, and make the reusable function in the competition index, but this is ok 👌

**Review by yoslie — APPROVED** [review]

**Review by agunghide — APPROVED** [review]

File: components/Competition/Banner/index.vue (hunk 1 above)
Comment (doohanas) [reply components/Competition/Banner/index.vue:44]: It's a nice solution, Thanks ⭐ . I have another approach to creating this.

First, in `pages/sayembara.vue` I can add conditional. See picture below.
showModal

Then, in `components/Competition/Join/index.vue` on the `onClickCTA` method I pass the second argument to provide the value `true` to show modal and `false` to open questionnaire page
onClickCTA

And what do you think❓ 
@yoslie

Comment (yoslie) [reply components/Competition/Banner/index.vue:44]: Sure this is ok, but from the function name we can see from the reading side showModal means to open modal, but questionnaire is a page.. 
so maybe either renaming the function (make it more global) or make a new function is the solution #cmiiw

Comment (doohanas) [reply components/Competition/Banner/index.vue:44]: Yes. it's quite confusing to rename the function to open the modal and the questionnaire page. I think it's good to create a new method to open the questionnaire page in `pages/sayembara.vue`. Okay, thanks for the discussion 🌟

**Pull request merged**